### PR TITLE
Update codeowners

### PR DESCRIPTION
**Pull request by dkistner**
**How to categorize this PR?**

/kind enhancement
/priority normal

**What this PR does / why we need it**:
Add extension maintainers as codeowners for the respective Terraform provider plugin configurations.

**Release note**:

```NONE

```

/invite @timebertt  @ialidzhikov @vpnachev
/cc @kon-angelo @MartinWeindel @prashanth26 @rfranzke @EmoinLanyu @jia-jerry

### CODE REVIEW

File: CODEOWNERS
Comment (vpnachev) [inline CODEOWNERS:4]:
```suggestion
/build/all/**       @gardener/gardener-maintainers @gardener-extension-provider-alicloud-maintainers @gardener-extension-provider-aws-maintainers @gardener-extension-provider-azure-maintainers @gardener-extension-provider-gcp-maintainers @gardener-extension-provider-openstack-maintainers
```

**Review by vpnachev — CHANGES_REQUESTED** [review]

File: CODEOWNERS
Comment (dkistner) [reply CODEOWNERS]: done

**Review by vpnachev — APPROVED** [review]
/lgtm
/squash

**Review by timebertt — APPROVED** [review]
/lgtm

**Comment by dkistner** [comment]
Can we go on and merge this one?

**Comment by vpnachev** [comment]
Please, ensure the PR checks are successful. Until then we cannot merge this PR.

**Pull request merged**
